v0/destinations/ga4/transform.js

```diff
@@ -84,410 +89,129 @@ function trackResponseBuilder(message, { Config }) {
-          )
-        );
-        break;
-      /* Sharing Section */
-      case "product_shared":
-        payload.name = eventNameMapping[event.toLowerCase()];
-        payload.params = constructPayload(
-          message,
-          mappingConfig[ConfigCategory.PRODUCT_SHARED.name]
-        );
-        payload.params = extractCustomFields(
-          message,
-          payload.params,
-          ["properties"],
-          getExclusionList(mappingConfig[ConfigCategory.PRODUCT_SHARED.name])
-        );
-        break;
-      case "cart_shared":
-        payload.name = eventNameMapping[event.toLowerCase()];
-        payload.params = constructPayload(
-          message,
-          mappingConfig[ConfigCategory.CART_SHARED.name]
-        );
-        payload.params = extractCustomFields(
-          message,
-          payload.params,
-          ["properties"],
-          getExclusionList(mappingConfig[ConfigCategory.CART_SHARED.name])
-        );
-        break;
-      /* Group */
-      case "group":
-        payload.name = eventNameMapping[event.toLowerCase()];
-        payload.params = constructPayload(
-          message,
-          mappingConfig[ConfigCategory.GROUP.name]
-        );
-        payload.params = extractCustomFields(
-          message,
-          payload.params,
-          ["properties"],
-          getExclusionList(mappingConfig[ConfigCategory.GROUP.name])
-        );
-        break;
-      /* GA4 Events */
-      case "earn_virtual_currency":
-        payload.name = eventNameMapping[event.toLowerCase()];
-        payload.params = constructPayload(
-          message,
-          mappingConfig[ConfigCategory.EARN_VIRTUAL_CURRENCY.name]
-        );
-        payload.params = extractCustomFields(
-          message,
-          payload.params,
-          ["properties"],
-          getExclusionList(
-            mappingConfig[ConfigCategory.EARN_VIRTUAL_CURRENCY.name]
-          )
-        );
-        break;
-      case "generate_lead":
-        payload.name = eventNameMapping[event.toLowerCase()];
-        payload.params = constructPayload(
-          message,
-          mappingConfig[ConfigCategory.GENERATE_LEAD.name]
-        );
-        payload.params = extractCustomFields(
-          message,
-          payload.params,
-          ["properties"],
-          getExclusionList(mappingConfig[ConfigCategory.GENERATE_LEAD.name])
-        );
-        break;
-      case "level_up":
-        payload.name = eventNameMapping[event.toLowerCase()];
-        payload.params = constructPayload(
-          message,
-          mappingConfig[ConfigCategory.LEVEL_UP.name]
-        );
-        payload.params = extractCustomFields(
-          message,
-          payload.params,
-          ["properties"],
-          getExclusionList(mappingConfig[ConfigCategory.LEVEL_UP.name])
-        );
-        break;
+
+    payload.name = ConfigCategory[`${event.toUpperCase()}`].event;
+
+    if (mappingConfig[ConfigCategory[`${event.toUpperCase()}`].name]) {
```

Comment:
We are already storing all the mapping files for all the config category is this check required? Can you explain a bit here